File: 2023.2/lab1/python/serial/find.py
from matrix import generate_matrix, max, min
import sys

if __name__ == "__main__":
    if len(sys.argv) != 2:
        print("Use: python3 find.py <size>")
        sys.exit(1)

    try:
        size = int(sys.argv[1])
    except ValueError:
        print("The size of the matrix must be an integer.")
        sys.exit(1)

    matrix = generate_matrix(size)
    print(f'Max value: {max(matrix)}')
    print(f'Min value: {min(matrix)}')

File: 2023.2/lab1/java/serial/Find.java
public class Find {
    public static void main(String[] args) {
        if (args.length != 1) {
            System.out.println("Use: java Main <size>");
            System.exit(1);
        }

        try {
            int size = Integer.parseInt(args[0]);

            int[][] matrix = Matrix.generateMatrix(size);

            System.out.printf("Max value: %d\n", Matrix.Max(matrix));
            System.out.printf("Min value: %d\n", Matrix.Min(matrix));
            
        } catch (NumberFormatException e) {
            System.out.println("The size of the matrix must be an integer.");
            System.exit(1);
        }

        
    }
}
